pkg/rsstorage/servers/s3server/s3_enumeration.go

```diff
@@ -157,6 +158,114 @@ func (a *DefaultAwsOps) BucketObjects(bucket, s3Prefix string, concurrency int,
 	}
 }
 
+func (a *DefaultAwsOps) BucketObjectsMap(bucket, s3Prefix string, concurrency int, recursive bool, reg *regexp.Regexp) (map[string]string, error) {
```

Comment:
No need to review this whole method in detail. I'll annotate what changed, apart from the signature.